### DISCUSSION TITLE
Android Proguard File Creation With No Upload to VPN Self Hosted Sentry

### DISCUSSION THREAD

**Discussion by btseytlinTCP** [getsentry/sentry-java#1912] [Q&A] [answered]
Hello!

We're using a self hosted Sentry instance behind a VPN, so automatic mapping upload fails due to the server/ports not being accessible from Github actions (where we run our assembling of production build).

Looking at the Sentry Gradle options, I'm not seeing one where the mapping is created but NOT uploaded. I'm looking for a scenario where the build process creates the mapping.txt file but is just sitting there so that I can action on it with a different build step. However, with [Sentry Android Gradle Plugin](https://github.com/getsentry/sentry-android-gradle-plugin) i am only able to either
- Create mapping + upload (autoUploadProguardMapping = true) where the upload fails thus failing the build
- or No mapping happens (autoUploadProguardMapping = false)

Is there a configuration within the Sentry Android Gradle Plugin where I can achieve this desired output? It's feeling like I need to do this with Sentry CLI but it would be nice if Gradle Plugin supported this use case.

Thank you!

(I didn't see discussions on the android gradle plugin repo so I started it here, rather than starting a "bug" there)

**Comment by romtsn** [comment, marked as answer]
Actually we are not the ones who's creating the mapping file, but the Android Gradle plugin. So your mapping file is always available (regardless of what error monitoring solution you're using), precisely at `build/outputs/mapping/$variantName/mapping.txt`.

When using `autoUploadProguardMapping=false` (note, that this is only available in the version `3.0.0` or above, which is currently at `3.0.0-beta.4`),  we do not auto-upload the mapping.txt file, but just generate the `uuid`, which is packaged inside your final apk/aab. This uuid is then used by our sentry-android SDK to associate all errors/events with the mapping file on the backend.

You can access this `uuid` under `build/generated/assets/sentry/$variantName/sentry-debug-meta.properties` and use it to manually upload the mapping file via sentry-cli (or an API call).

I think this should be sufficient for your case, let me know if we can assist further!

**Reply by btseytlinTCP** [reply]
>Actually we are not the ones who's creating the mapping file, but the Android Gradle plugin. So your mapping file is always available (regardless of what error monitoring solution you're using), precisely at build/outputs/mapping/$variantName/mapping.txt.

Thank you for the quick answer, I've been hunting down the solution since this tidbit that you provided! We needed to stand up our own separate infrastructure (pulling the mapping.txt and sentry-debug-meta.properties files into an S3 bucket and then running sentry CLI from there). That helped us get around the scenario we were in (self hosted Sentry behind a VPN, with the build process being in Github Action which doesn't have VPN access). 

For us, the file location was slightly different: `./app/build/outputs/mapping/$variantName/mapping.txt`. This might be because we are running this in a github action or because it's specific to our repo. I was able to chase down that location by running a command within the github action to find the file: `find . -name mapping.txt`

Thanks again!

**Reply by marandaneto** [reply]
If you don't want to add the file `sentry-debug-meta.properties` in the assets folder of your App, you can also add a config entry to the Manifest file, https://github.com/getsentry/sentry-docs/issues/4431

**Reply by btseytlinTCP** [reply]
Is that to say that I can set the uuid there and then just use that same value where I need it?

**Reply by marandaneto** [reply]
yes, but the `uuid` should match the uploaded mapping file via sentry-cli, its just an alternative way of setting it instead of sentry-debug-meta.properties

**Comment by romtsn** [comment]
(also, I just enabled discussions on the gradle plugin repo, thanks for the heads up)

**Comment by marandaneto** [comment]
Since the Plugin uses sentry-cli under the hood, you can export an env. var. or set the `url` in the `sentry.properties` file.
https://docs.sentry.io/product/cli/configuration/#configuration-file

Example using `sentry.properties`: https://github.com/getsentry/sentry-react-native/blob/main/sample/android/sentry.properties#L1
or `export SENTRY_URL=https://sentry.io/`
Using your Server URL, of course.
